### DISCUSSION TITLE
Rerun from failure after deploying new code

### DISCUSSION THREAD

**Discussion by saaramitay** [dagster-io/dagster#18620] [Q&A]
Hi,

Our way of work with Dagster is deployments on K8s in GCP, sensors that are waiting for trigger to spin a new pod containing the actual pipeline to run.

Sometimes the pipeline fails at some op because of a code issue, we then need to fix the code, deploy the pipeline with the updated code and run again.

Wanted to know if it is possible after deploying the updated code to rerun from failure the failed pipeline (so it will be able to keep the results of the succeeded previous ops and not run them again) without the need to create a new run?

**Comment by jamiedemaria** [comment]
As long as you are storing the outputs from the upstream/successful ops using an I/O manager configured for persistent storage (ie a GCS bucket rather than a temporary file) then you should be able to rerun from failure and Dagster will use the I/O manager to fetch the results from the successful previous ops

**Reply by saaramitay** [reply]
Hi @jamiedemaria thanks for the reply.

Do you have maybe an example you can share regarding how to implement it and what it requires from the helm chart of dagster / the code of the pipelines?

**Reply by jamiedemaria** [reply]
It shouldn't require anything other than using a suitable I/O manager (if you are using an I/O manager). Then you'll be able to retry from failure in the UI

**Reply by saaramitay** [reply]
im looking to have automatic retries
